rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/NarayanaLRARecovery.java

```diff
@@ -60,48 +64,86 @@ public void waitForCallbacks(URI lraId) {
 
     @Override
     public boolean waitForEndPhaseReplay(URI lraId) {
-        String host = lraId.getHost();
-        int port = lraId.getPort();
-        if (!recoverLRAs(host, port, lraId)) {
+        if (!recoverLRAs(lraId)) {
             // first recovery scan probably collided with periodic recovery which started
             // before the test execution so try once more
-            return recoverLRAs(host, port, lraId);
+            return recoverLRAs(lraId);
         }
-
         return true;
     }
 
     /**
-     * Invokes LRA coordinator recovery REST endpoint and returns whether the recovery of intended LRAs happended
+     * Invokes LRA coordinator recovery REST endpoint and returns whether the recovery of intended LRAs happened
      *
-     * @param host  the LRA coordinator host address
-     * @param port  the LRA coordinator port
      * @param lraId the LRA id of the LRA that is intended to be recovered
      * @return true the intended LRA recovered, false otherwise
      */
-    private boolean recoverLRAs(String host, int port, URI lraId) {
+    private boolean recoverLRAs(URI lraId) {
         // trigger a recovery scan
         Client recoveryCoordinatorClient = ClientBuilder.newClient();
 
         try {
-            String recoveryCoordinatorUrl = String.format("http://%s:%d/%s/recovery",
-                host, port, LRAConstants.RECOVERY_COORDINATOR_PATH_NAME);
-            WebTarget recoveryTarget = recoveryCoordinatorClient.target(URI.create(recoveryCoordinatorUrl));
+            URI lraCoordinatorUri = extractLRACoordinatorUri(lraId);
+            URI recoveryCoordinatorUri = UriBuilder.fromUri(lraCoordinatorUri)
+                    .path(RECOVERY_COORDINATOR_PATH_NAME).build();
+            WebTarget recoveryTarget = recoveryCoordinatorClient.target(recoveryCoordinatorUri);
 
             // send the request to the recovery coordinator
             Response response = recoveryTarget.request().get();
             String json = response.readEntity(String.class);
             response.close();
 
-            if (json.contains(lraId.toASCIIString())) {
-                // intended LRA didn't recover
-                return false;
-            }
-
-            return true;
+            // intended LRA didn't recover
```

Comment:
The intend is to make the path dynamic. When the coordinator is run with Quarkus the path is explicit. But in case of e.g. WildFly the path depends on the deployment name. The deployment name may differ for every test.

Reply:
oh, I'm sorry for that. I misunderstood your point. When you consider prior code cleaner I will change it back.